repose-aggregator/core/repose-core/src/main/java/org/openrepose/nodeservice/ReqResHeaderService.java

```diff
@@ -17,27 +17,21 @@
  * limitations under the License.
  * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
  */
-package org.openrepose.nodeservice.response;
+package org.openrepose.nodeservice;
 
-import org.openrepose.commons.utils.StringUtilities;
-import org.openrepose.core.services.headers.common.ViaHeaderBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.LoggerFactory;
 
 import javax.servlet.http.HttpServletRequest;
 
-public class ViaResponseHeaderBuilder extends ViaHeaderBuilder {
-
-    private final String reposeVersion;
-    private final String viaReceivedBy;
-
-    public ViaResponseHeaderBuilder(String reposeVersion, String configuredViaReceivedBy) {
-        this.reposeVersion = reposeVersion;
-        this.viaReceivedBy = StringUtilities.isBlank(configuredViaReceivedBy) ? "Repose" : configuredViaReceivedBy;
-    }
-
-    @Override
-    protected String getViaValue(HttpServletRequest request) {
-        StringBuilder builder = new StringBuilder(" ");
-
-        return builder.append(viaReceivedBy).append(" (Repose/").append(reposeVersion).append(")").toString();
+public interface ReqResHeaderService {
+    default void appendProtocolVersion(final StringBuilder builder, final HttpServletRequest request) {
```

Comment:
That interesting, but I can't say I am overly surprised. The Servlet specification always seemed to want to be general, but as a result, does not provide the best support for HTTP (or body manipulation, or response manipulation).

I guess for now we should keep current behavior, even if it is wrong. I'm not sure what the correct solution would be, given that we don't seem to have access to the information we need from the response. I wonder if perhaps we can get that information from the Apache client when we make the request to the origin service and stick it in a response parameter or something like that. That's neither here nor there, though.